red.py

```diff
@@ -520,53 +522,54 @@ def load_cogs():
         print("\nFailed to load: {}\n".format(" ".join(failed)))
 
 
-def main():
+def main(bot):
     check_folders()
-    if not settings.no_prompt:
-        interactive_setup()
-    load_cogs()
+    if not bot.settings.no_prompt:
+        interactive_setup(bot.settings)
+    load_cogs(bot)
 
     print("Logging into Discord...")
     bot.uptime = datetime.datetime.utcnow()
 
-    if settings.login_credentials:
-        yield from bot.login(*settings.login_credentials,
-                             bot=not settings.self_bot)
+    if bot.settings.login_credentials:
+        yield from bot.login(*bot.settings.login_credentials,
+                             bot=not bot.settings.self_bot)
     else:
         print("No credentials available to login.")
         raise RuntimeError()
     yield from bot.connect()
 
+
 if __name__ == '__main__':
     sys.stdout = TextIOWrapper(sys.stdout.detach(),
                                encoding=sys.stdout.encoding,
                                errors="replace",
                                line_buffering=True)
-    set_logger()
+    bot = initialize()
     error = False
     loop = asyncio.get_event_loop()
     try:
-        loop.run_until_complete(main())
+        loop.run_until_complete(main(bot))
     except discord.LoginFailure:
         error = True
-        logger.error(traceback.format_exc())
-        if not settings.no_prompt:
-            choice = input("Invalid login credentials. "
-                           "If they worked before Discord might be having temporary "
-                           "technical issues.\nIn this case, press enter and "
-                           "try again later.\nOtherwise you can type 'reset' to "
-                           "reset the current credentials and set them "
-                           "again the next start.\n> ")
+        bot.logger.error(traceback.format_exc())
+        if not bot.settings.no_prompt:
+            choice = input("Invalid login credentials. If they worked before "
+                           "Discord might be having temporary technical "
+                           "issues.\nIn this case, press enter and try again "
+                           "later.\nOtherwise you can type 'reset' to reset "
+                           "the current credentials and set them again the "
+                           "next start.\n> ")
             if choice.lower().strip() == "reset":
-                settings.token = None
-                settings.email = None
-                settings.password = None
-                settings.save_settings()
+                bot.settings.token = None
+                bot.settings.email = None
+                bot.settings.password = None
+                bot.settings.save_settings()
     except KeyboardInterrupt:
         loop.run_until_complete(bot.logout())
     except:
         error = True
-        logger.error(traceback.format_exc())
+        bot.logger.error(traceback.format_exc())
```

Comment:
I'll change it, but I didn't write that bit before. Only made it work with my other changes.